bin/setup.csh

```diff
@@ -0,0 +1,46 @@
+# setup lsstsw environment
+#
+# source this file from your ~/.cshrc
+
+
+set cmd=($_)        # possibly blank, but can't be parsed as $_
+if ( $?0 ) then        # direct execution
+    set source=$0
+else if ( $#cmd >= 3 ) then        # direct sourcing
+    set source=${cmd[2]}
+else if (-f /usr/sbin/lsof ) then        # indirect sourcing
+    set source=`/usr/sbin/lsof +p $$ | grep -oE /.\*setup.csh`
+endif
+unset cmd
+
+if ( $?source ) then
+    set LSSTSW=`dirname $source`
+    set LSSTSW=`cd $LSSTSW/.. && pwd`
+endif
+
+if ( ! $?LSSTSW ) then
+    echo "error: could not figure out LSSTSW directory"
+    echo '  you can specify the directory by setting $LSSTW in your ~/.cshrc'
+    exit 1
```

Comment:
Yeah, that makes me nervous too. But it seems to work (both sourced and executed), and I don't know of any other equivalents to bash `return`.